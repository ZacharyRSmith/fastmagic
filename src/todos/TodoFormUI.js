import React from 'react';
import { Button, FormControl, FormGroup, ControlLabel, HelpBlock } from 'react-bootstrap';

const RequiredTextFormGroup = ({ field, fieldState, handleChange }) => {
  return (
    <FormGroup {...{
      controlId: field.name,
      validationState: fieldState.validationState || null
    }}>
      <ControlLabel>{field.label}</ControlLabel>
      <FormControl {...{
        componentClass: field.componentClass || 'input',
        onChange: e => { e.persist(); handleChange(e) },
        type: 'text', // relevant only if this FormControl's componentClass is 'input'
        value: fieldState.value
      }}/>
      <FormControl.Feedback/>
      <HelpBlock {...{
        bsClass: fieldState.validationState ? 'help-block show' : 'hide'
      }}>
        {field.helpMessage}
      </HelpBlock>
    </FormGroup>
  );
};

const TodoFormUI = props => {
  return (
    <form onSubmit={props.handleSubmit}>
      <RequiredTextFormGroup {...{
        field: {
          name: 'text',
          label: 'Text',
          helpMessage: 'Please provide todo\'s text'
        },
        fieldState: props.formData['text'],
        handleChange: props.handleChange
      }}/>
      <RequiredTextFormGroup {...{
        field: {
          name: 'owner',
          label: 'Owner',
          helpMessage: 'Please provide name of todo\'s owner'
        },
        fieldState: props.formData['owner'],
        handleChange: props.handleChange
      }}/>

      <Button {...{ className: 'submit', type: 'submit' }}>Create</Button>
    </form>
  );
};

export default TodoFormUI;
