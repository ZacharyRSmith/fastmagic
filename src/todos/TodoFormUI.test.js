import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import TodoFormUI from './TodoFormUI';

it('renders without crashing', () => {
  const wrapper = shallow(
    <TodoFormUI {...{
      formData: {},
      handleChange: () => {},
      handleSubmit: () => {}
    }} />
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});
