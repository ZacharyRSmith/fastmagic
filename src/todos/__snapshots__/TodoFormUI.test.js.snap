exports[`test renders without crashing 1`] = `
<form
  onSubmit={[Function]}>
  <RequiredTextFormGroup
    field={
      Object {
        "helpMessage": "Please provide todo\'s text",
        "label": "Text",
        "name": "text",
      }
    }
    handleChange={[Function]} />
  <RequiredTextFormGroup
    field={
      Object {
        "helpMessage": "Please provide name of todo\'s owner",
        "label": "Owner",
        "name": "owner",
      }
    }
    handleChange={[Function]} />
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    className="submit"
    disabled={false}
    type="submit">
    Create
  </Button>
</form>
`;
